subprojects/core/src/main/java/org/gradle/api/internal/tasks/SnapshotTaskInputsOperationDetails.java

```diff
@@ -28,20 +30,19 @@
  *
  * This operation is executed only when the build cache is enabled.
  *
- * This class is intentionally internal and consumed by the build scan plugin.
- *
  * @since 4.0
  */
+@UsedByScanPlugin
 public final class SnapshotTaskInputsOperationDetails implements BuildOperationDetails<SnapshotTaskInputsOperationDetails.Result> {
 
-    private final String taskPath;
+    private final Task task;
 
-    public SnapshotTaskInputsOperationDetails(String taskPath) {
-        this.taskPath = taskPath;
+    public SnapshotTaskInputsOperationDetails(Task task) {
+        this.task = task;
     }
 
-    public String getTaskPath() {
-        return taskPath;
+    public Task getTask() {
```

Comment:
@oehme Task identity paths are not guaranteed to be unique. Gradle will happily execute two tasks of the same project with the same name. This is why we have to key off the task object. It's the only thing that's guaranteed to be unique. 

Try this build script and run `all`:

```
task all

task t {
  doFirst {
    println "first"
  }
}

all.dependsOn t

task t(overwrite: true) {
  doFirst {
    println "second"
  }
}

task intermediary {
  dependsOn t
}

all.dependsOn intermediary
```

This is not as uncommon as you might think, and caused us problems at the start of build scans.